File: Python/min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost):
        # dynamic programming method
        # so difficult, copy from discussion
        _array = [0] * len(cost)
        for idx in range(2, len(cost)):
            _array[idx] = min(_array[idx-1] + cost[idx-1], _array[idx-2] + cost[idx-2])
        return min(_array[-1] + cost[-1], _array[-2] + cost[-2])

File: C/min-cost-climbing-stairs.c
int minCostClimbingStairs(int* cost, int costSize){
    int i;
    int dp[costSize + 1];

    for (i = 0; i < costSize + 1; ++i) {
        dp[i] = 0;
    }
    for (i = 2; i < costSize + 1; ++i) {
        if (dp[i - 2] + cost[i - 2] < dp[i - 1] + cost[i - 1]) dp[i] = dp[i - 2] + cost[i - 2];
        else dp[i] = dp[i - 1] + cost[i - 1];
    }
    return dp[costSize];
}
